admin_guide/configuring_authentication.adoc

```diff
@@ -0,0 +1,533 @@
+
+----
+oauthConfig:
+  ...
+  identityProviders:
+  - name: my_request_header_provider <1>
+    challenge: false <2>
+    login: false <3>
+    provider:
+      apiVersion: v1
+      kind: RequestHeaderIdentityProvider
+      clientCA: /path/to/client-ca.file <4>
+      headers: <5>
+      - X-Remote-User
+      - SSO-User
+----
+<1> This provider name is prefixed to the user name in the request header to form an identity name.
+<2> `*RequestHeaderIdentityProvider*` cannot be used to send `WWW-Authenticate` challenges.
+<3> `*RequestHeaderIdentityProvider*` cannot be used to back a login page.
+<4> Optional: PEM-encoded certificate bundle. If set, a valid client certificate must be presented and validated against the certificate authorities in the specified file before the request headers are checked for user names.
+<5> Header names to check, in order, for user names. The first header containing a value is used as the user name. Required, case-insensitive.
+====
+
+=== GitHub [[GitHub]]
+
+Set `*GitHubIdentityProvider*` in the `*identityProviders*` stanza to use
+https://github.com/[GitHub] as an identity provider, using the
+https://developer.github.com/v3/oauth/[OAuth integration].
+
+NOTE: Using GitHub as an identity provider requires users to get a token using `_<master>_/oauth/token/request` to use with command-line tools.
+
+.Master Configuration Using `*GitHubIdentityProvider*`
+====
+
+----
+oauthConfig:
+  ...
+  identityProviders:
+  - name: github <1>
+    challenge: false <2>
+    login: true <3>
+    provider:
+      apiVersion: v1
+      kind: GitHubIdentityProvider
+      clientID: ... <4>
+      clientSecret: ... <5>
+----
+<1> This provider name is prefixed to the GitHub numeric user ID to form an identity name. It is also used to build the callback URL.
+<2> `*GitHubIdentityProvider*` cannot be used to send `WWW-Authenticate` challenges.
+<3> When `true`, unauthenticated token requests from web clients (like the Management Console) are redirected to GitHub to log in.
+<4> The client ID of a link:https://github.com/settings/applications/new[registered GitHub OAuth application]. The application must be configured with a callback URL of `_<master>_/oauth2callback/_<identityProviderName>_`.
+<5> The client secret issued by GitHub.
+====
+
+=== Google [[Google]]
+
+Set `*GoogleIdentityProvider*` in the `*identityProviders*` stanza to use Google as
+an identity provider, using
+https://developers.google.com/identity/protocols/OpenIDConnect[Google's OpenID
+Connect integration].
+
+NOTE: Using Google as an identity provider requires users to get a token using
+`_<master>_/oauth/token/request` to use with command-line tools.
+
+.Master Configuration Using `*GoogleIdentityProvider*`
+====
+
+----
+oauthConfig:
+  ...
+  identityProviders:
+  - name: google <1>
+    challenge: false <2>
+    login: true <3>
+    provider:
+      apiVersion: v1
+      kind: GoogleIdentityProvider
+      clientID: ... <4>
+      clientSecret: ... <5>
```

Comment:
missing the `hostedDomain` bits added in https://github.com/openshift/openshift-docs/pull/306